src/picker_gui.cpp

```diff
@@ -64,29 +65,53 @@ PickerCallbacks::~PickerCallbacks()
  */
 static void PickerLoadConfig(const IniFile &ini, PickerCallbacks &callbacks)
 {
-	const IniGroup *group = ini.GetGroup(callbacks.ini_group);
-	if (group == nullptr) return;
+	/* Create a set of all the collections to load */
+	std::set<std::string> collections;
+	std::string type = callbacks.ini_group + "-";
+	size_t size = type.size();
+	collections.insert(callbacks.ini_group);
+
+	for (auto it = ini.groups.begin(); it != ini.groups.end(); it++) {
+		std::string_view name = it->name;
+		if (name.find(type) != 0) continue;
+		if (name.size() > size) collections.insert(it->name);
+	}
 
 	callbacks.saved.clear();
-	for (const IniItem &item : group->items) {
-		std::array<uint8_t, 4> grfid_buf;
+	for (auto it = collections.begin(); it != collections.end(); it++) {
+		const IniGroup *group = ini.GetGroup(*it);
+		if (group == nullptr) continue;
+
+		/* Read the collection name */
+		auto collection_pos = (*it).find('-');
+		std::string name = GetString(STR_PICKER_DEFAULT_COLLECTION);
+		if (collection_pos != std::string_view::npos) name = (*it).substr(collection_pos + 1);
+
+		if (group->items.empty() && collection_pos != std::string_view::npos) {
+			callbacks.saved[name];
+			continue;
+		}
```

Comment:
I think there's actually no need to keep a list of each collection, you can parse as you go.

Something like
```c++
for (const IniGroup &group : ini.groups) {
    if (!group.name.starts_with(callbacks.ini_group));
    auto pos = group.name.find('-');
    std::string collection = (pos == std::string_view::npos) ? "" : group.name.substr(pos + 1);

    // ... rest of ini file loading ...

}
```

You might want to validate that when `pos == std::string_view::npos`, `group.name == callbacks.ini_group` as well, too.